### PR TITLE
TiUP: Support setting CPUAffinity

### LINKED ISSUE #1927: Support setting CPUAffinity through TiUP
## Feature Request

**Is your feature request related to a problem? Please describe:**

Sometimes I have a server with many cores, so I want to deploy multiple TiDB or TiKV nodes on one server. And I hope different nodes on the same server don't interfere with each other.

It's still a luxury to deploy only one process on one NUMA node. But TiUP does not provide a way to isolate processes in a more fine-grained mode.

**Describe the feature you'd like:**

Systemd provides `CPUAffinity` option. I hope TiUP cluster can integrate this feature to set CPU affinity through cluster configuration file.

**Why the featue is needed:**

Have described my scenario above :)

**Describe alternatives you've considered:**

Now I need to patch the systemd service manually.

**Teachability, Documentation, Adoption, Migration Strategy:**

Doc for `CPUAffinity`: https://www.freedesktop.org/software/systemd/man/systemd-system.conf.html

### PR DESCRIPTION
**Pull request by YaozhengWang**
### What problem does this PR solve? 

- close #1927

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
NONE
```

### CODE REVIEW

**Review by nexustar — APPROVED** [review]

File: embed/templates/scripts/run_tidb.sh.tpl [hunk 1]
```diff
@@ -17,7 +17,9 @@ cd "${DEPLOY_DIR}" || exit 1
   {{- end}}
 {{- end}}
 
-{{- if .NumaNode}}
+{{- if and .NumaNode .NumaCores}}
```
Comment (AstroProfundis) [inline embed/templates/scripts/run_tidb.sh.tpl:20]: Does specifying `NumaCores` without `NumaNode` make any sence?

Comment (YaozhengWang) [reply embed/templates/scripts/run_tidb.sh.tpl:20]: The user must specify NumaNode first, and then specify NumaCores，

File: pkg/cluster/spec/tiflash.go [hunk 2]
```diff
@@ -58,6 +58,7 @@ type TiFlashSpec struct {
 	TmpDir               string                 `yaml:"tmp_path,omitempty"`
 	Offline              bool                   `yaml:"offline,omitempty"`
 	NumaNode             string                 `yaml:"numa_node,omitempty" validate:"numa_node:editable"`
+	NumaCores            string                 `yaml:"numa_cores,omitempty" validate:"numa_cores:editable"`
```
Comment (nexustar) [inline pkg/cluster/spec/tiflash.go:61]: why not numa_core

**Pull request closed without merging**